opensearch-operator/pkg/reconcilers/securityconfig.go

```diff
@@ -64,75 +64,46 @@ func (r *SecurityconfigReconciler) Reconcile() (ctrl.Result, error) {
 	}
 
 	var configSecretName string
+	var checksumval string
 	adminCertName := r.determineAdminSecret()
 	namespace := r.instance.Namespace
 	clusterName := r.instance.Name
-	//Checking if Security Config values are empty and creates a default-securityconfig secret
-	if r.instance.Spec.Security.Config == nil || r.instance.Spec.Security.Config.SecurityconfigSecret.Name == "" {
-		r.logger.Info(clusterName + "-default-securityconfig is being created")
-		SecurityConfigSecretName := clusterName + "-default-securityconfig"
-		//Basic SecurityConfigSecret secret with default settings
-		SecurityConfigSecret := corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: SecurityConfigSecretName, Namespace: namespace}, Type: corev1.SecretTypeOpaque, StringData: make(map[string]string)}
-		if err := r.Get(r.ctx, client.ObjectKey{Name: SecurityConfigSecretName, Namespace: namespace}, &SecurityConfigSecret); err == nil {
-			r.logger.Info(clusterName + "-default-securityconfig secret exists")
-		} else {
-			r.logger.Info("creating " + clusterName + "-default-securityconfig secret")
-			//Reads all securityconfig files and adds them to secret Stringdata
-			files, err := ioutil.ReadDir("./helperfiles/defaultsecurityconfigs/")
-			if err != nil {
-				r.logger.Info("Failed to read directory helperfiles/defaultsecurityconfigs")
-				//panic(err)
-			}
-			for _, f := range files {
-				fileBytes, err := ioutil.ReadFile("./helperfiles/defaultsecurityconfigs/" + f.Name())
-				if err != nil {
-					r.logger.Info("Failed to add " + f.Name() + clusterName + "-default-securityconfig secret")
-					//panic(err)
-				}
-				SecurityConfigSecret.StringData[f.Name()] = string(fileBytes)
-			}
-			if err := ctrl.SetControllerReference(r.instance, &SecurityConfigSecret, r.Client.Scheme()); err != nil {
-				return ctrl.Result{}, err
-			}
-			//r.Create(r.ctx, &SecurityConfigSecret)
-			if err := r.Create(r.ctx, &SecurityConfigSecret); err != nil {
-				r.logger.Error(err, "Failed to create default"+clusterName+"-default-securityconfig secret")
-				return ctrl.Result{}, err
-			}
-
-		}
-		configSecretName = clusterName + "-default-securityconfig"
-	} else {
-		//Use a user passed value of SecurityconfigSecret name
-		configSecretName = r.instance.Spec.Security.Config.SecurityconfigSecret.Name
-	}
-
 	jobName := clusterName + "-securityconfig-update"
-
 	if adminCertName == "" {
 		r.logger.Info("Cluster is running with demo certificates.")
 		return ctrl.Result{}, nil
 	}
+	//Checking if Security Config values are empty and creates a default-securityconfig secret
+	if r.instance.Spec.Security.Config == nil || r.instance.Spec.Security.Config.SecurityconfigSecret.Name == "" {
+		r.logger.Info("No user passed SecurityconfigSecret")
```

Comment:
This part seems a little superfluous, we can probably just invert the if statement and get rid of this log line.

Reply:
So I don't think we need to log an informational message that we're not doing anything.  If we do that should be in the `else` clause as the major bard of the logic is what to do when the SecurityConfig isn't nil.  So this block should be
```go
if r.instance.Spec.Security.Config != nil && r.instance.Spec.Security.Config.SecurityconfigSecret.Name != "" {
    do the stuff
}
```
Then if we do want to log something it should probably be at debug, so we can add
```
else {
    log debug message
}
```

Reply:
Pushed in my latest commit @dbason